nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -204,6 +204,20 @@ def get_profiles() -> List[str]:
     else:
         return []
 
+def should_update(v_from: str, v_to: str) -> int:
+    # see https://github.com/systemd/systemd/blob/main/src/boot/bootctl.c compare_product function
+
+    len_from = len(v_from)
+    len_to = len(v_to)
+
+    if len_from < len_to:
+        return False
```

Comment:
The type hint is wrong if we return a boolean here.

Reply:
Dang, sure, I'll fix it when I have a keyboard in front of me.